analyzer/codechecker_analyzer/analyzers/gcc/analyzer.py

```diff
@@ -0,0 +1,255 @@
+        This is mostly used for CTU, which is absent in GCC.
+        """
+        pass
+
+    def construct_analyzer_cmd(self, result_handler):
+        """
+        Construct analyzer command for gcc.
+        """
+        # TODO: This is not a try-catch block, like the other analyzers. Why
+        # should it? Should the others be? When can list creating list to have
+        # unforeseen exceptions where a general catch is justified?
+        config = self.config_handler
+
+        # We don't want GCC do start linking, but -fsyntax-only stops the
+        # compilation process too early for proper diagnostics generation.
+        analyzer_cmd = [Gcc.analyzer_binary(), '-fanalyzer', '-c',
+                        '-o/dev/null']
+
+        # Add extra arguments.
+        analyzer_cmd.extend(config.analyzer_extra_arguments)
+
+        analyzer_cmd.extend(self.buildaction.analyzer_options)
+
+        analyzer_cmd.append('-fdiagnostics-format=sarif-file')
+
+        for checker_name, value in config.checks().items():
+            if value[0] == CheckerState.disabled:
+                # TODO python3.9 removeprefix method would be nicer
+                # than startswith and a hardcoded slicing
+                analyzer_cmd.append(
+                    codechecker_name_to_actual_name_disabled(checker_name))
+
+        compile_lang = self.buildaction.lang
+        if not has_flag('-x', analyzer_cmd):
+            analyzer_cmd.extend(['-x', compile_lang])
+
+        analyzer_cmd.append(self.source_file)
+
+        LOG.debug_analyzer("Running analysis command "
+                           f"'{shlex.join(analyzer_cmd)}'")
+
+        return analyzer_cmd
+
+    @classmethod
+    def get_analyzer_checkers(self):
+        """
+        Return the list of the supported checkers.
+        """
+        command = [self.analyzer_binary(), "--help=warning"]
+        checker_list = []
+
+        try:
+            output = subprocess.check_output(command)
+
+            # Still contains the help message we need to remove.
+            for entry in output.decode().split('\n'):
+                warning_name, _, description = entry.strip().partition(' ')
+                # GCC Static Analyzer names start with -Wanalyzer.
+                if warning_name.startswith('-Wanalyzer'):
+                    # Rename the checkers interally (similarly to how we
+                    # support cppcheck)
+                    renamed_checker_name = \
+                        actual_name_to_codechecker_name(warning_name)
+                    checker_list.append(
+                        (renamed_checker_name, description.strip()))
+            return checker_list
+        except (subprocess.CalledProcessError) as e:
+            LOG.error(e.stderr)
+        except (OSError) as e:
+            LOG.error(e.errno)
+        return []
+
+    @classmethod
+    def get_analyzer_config(cls):
+        """
+        Config options for gcc.
+        """
+        # TODO
+        return []
```

Comment:
To return an empty list is valid here I think.

Now, I am getting this error:
```
workspace/test-projects/xerces-c]$ CodeChecker analyzers --analyzer-config gcc
[ERROR 2023-10-16 12:00] - Failed to get analyzer configuration options for 'gcc' analyzer! Please try to upgrade your analyzer version to use this feature.
```
However not supporting any option is not an error.

CodeChecker should write something like this instead:

`No analyzer options supported for gcc.`